src/__swaps__/utils/flipAssets.ts

```diff
@@ -0,0 +1,140 @@
+  inputMethod: SharedValue<inputMethods>;
+  lastTypedInput: SharedValue<inputKeys>;
+  focusedInput: SharedValue<inputKeys>;
+}) => {
+  'worklet';
+  const inputNativePrice = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+  const outputNativePrice = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
+  const hasNonZeroInputPrice = greaterThanWorklet(inputNativePrice, 0);
+  const hasNonZeroOutputPrice = greaterThanWorklet(outputNativePrice, 0);
+
+  const inputBalance = internalSelectedInputAsset.value?.maxSwappableAmount || 0;
+  const hasInputBalance = greaterThanWorklet(inputBalance, 0);
+
+  const prevInputNativeValue = inputValues.value.inputNativeValue;
+
+  const outputBalance = internalSelectedOutputAsset.value?.maxSwappableAmount || 0;
+  const hasOutputBalance = greaterThanWorklet(outputBalance, 0);
+
+  const calculatedInputAmount: string | number = greaterThanWorklet(inputNativePrice, 0)
+    ? divWorklet(prevInputNativeValue, inputNativePrice)
+    : 0;
+  const newFormattedInputAmount = valueBasedDecimalFormatter({
+    amount: calculatedInputAmount,
+    nativePrice: inputNativePrice,
+    roundingMode: 'up',
+    isStablecoin: internalSelectedInputAsset.value?.type === 'stablecoin',
+    stripSeparators: true,
+  });
+
+  let newInputAmount: string | number = newFormattedInputAmount;
+  let newOutputAmount: string | number = 0;
+
+  const exceedsMaxBalance = greaterThanWorklet(newInputAmount, inputBalance);
+  const validBalanceIfAny = (hasInputBalance && !exceedsMaxBalance) || !hasInputBalance;
+
+  // determine if we previously had max selected and can still set max on the new input
+  const prevInputAmount = inputValues.value.inputAmount;
+  const setToMax = hasInputBalance && hasOutputBalance && equalWorklet(prevInputAmount, outputBalance);
+
+  if (hasNonZeroInputPrice && hasNonZeroOutputPrice && validBalanceIfAny && !setToMax) {
+    // use previous native input amount if available
+    const prevOutputNativeValue = inputValues.value.outputNativeValue;
+    newOutputAmount = divWorklet(prevOutputNativeValue, outputNativePrice);
+    inputMethod.value = 'inputAmount';
+  } else if (hasInputBalance && hasOutputBalance) {
+    // use slider position if available
+    const { inputAmount: inputAmountBasedOnSlider } = getInputValuesForSliderPositionWorklet({
+      selectedInputAsset: internalSelectedInputAsset.value,
+      percentageToSwap: percentageToSwap.value,
+      sliderXPosition: sliderXPosition.value,
+    });
+    newInputAmount = inputAmountBasedOnSlider;
+    inputMethod.value = 'slider';
+    lastTypedInput.value = 'inputAmount';
+    focusedInput.value = 'inputAmount';
+    inputValues.modify(values => {
+      return {
+        ...values,
+        inputAmount: newInputAmount,
+        inputNativeValue: mulWorklet(newInputAmount, inputNativePrice),
+      };
+    });
+    return;
```

Comment:
only for slider do we return early since we do not want to update the `outputAmount` and `outputNativeValue` like all the other flows; we just want to modify the `inputAmount` and `inputNativeValue`